apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -635,12 +647,22 @@ static void tc_pthread_pthread_create_exit_join(void)
 	TC_ASSERT_EQ("pthread_join", ret_chk, OK);
 	TC_ASSERT_EQ("pthread_exit", p_value, (void *)EDEADLK);
 
+	/* Wait to release the above pthread resource */
+	sleep(1);
+
 	ret_chk = pthread_join(pthread, NULL);
-	TC_ASSERT_EQ("pthread_join", ret_chk, EINVAL);
+	TC_ASSERT_EQ("pthread_join", ret_chk, ESRCH);
 
 	ret_chk = pthread_join(INVALID_PID, NULL);
 	TC_ASSERT_EQ("pthread_join", ret_chk, ESRCH);
 
+	task_pid = task_create("inval_task", 100, 1024, inval_task, NULL);
+	TC_ASSERT_GT("task_create", task_pid, 0);
+
+	ret_chk = pthread_join(task_pid, NULL);
+	TC_ASSERT_EQ("pthread_join", ret_chk, EINVAL);
+	inval_task_alive = false;
```

Comment:
How about using task_delete instead of controlling the task lifecycle using static variable? I think it is better to understand.